query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -262,6 +269,81 @@ private String convertOperator2String(Operator operator) {
     }
   }
 
+  private LiteralConstant[] getKeyValuePairForComplexAttribute(Expression lhs, Expression rhs) {
+    String pathExpression = lhs.getAttributeExpression().getSubpath();
+    LiteralConstant[] kvp = convertExpressionToMapLiterals(rhs);
+    kvp[0] =
+        LiteralConstant.newBuilder()
+            .setValue(Value.newBuilder().setString(pathExpression).build())
+            .build();
+    return kvp;
+  }
+
+  private String handleFilterForComplexAttribute(Filter filter, Builder paramsBuilder) {
+    LiteralConstant[] kvp = getKeyValuePairForComplexAttribute(filter.getLhs(), filter.getRhs());
+    String keyCol = convertExpressionToMapKeyColumn(filter.getLhs());
+    String valCol = convertExpressionToMapValueColumn(filter.getLhs());
+
+    if (!SUPPORTED_OPERATORS_FOR_MAP_ATTRIBUTES.contains(filter.getOperator())) {
```

Comment:
Similarly, we also have to check if the mapped attributedId (e.g Span.tags) maps to a complex column in Pinot. Only supported transformation should pass else we need to throw an exception. In our current case, the mapped column should be of type mapFields only - https://github.com/hypertrace/query-service/blob/main/query-service/src/main/resources/configs/common/application.conf#L58

`Span.tags` -> is of map type, if not we need to throw an exception.

Reply:
`handleFilterForComplexAttribute`
- check if the complex attribute is of map type (mapFields)
- if, yes, handleFilterForMapColumn

`handleFilterForMapColumn`
- check here that it has supported operators
- if not, throw an exception
- if, yes, do a transformation

Reply:
Currently, since the only complex attribute is the map attribute, keeping only `handleFilterForMapColumn` function

Reply:
resolved